Include/cpython/pyatomic_msc.h

```diff
@@ -0,0 +1,860 @@
+#ifndef Py_ATOMIC_MSC_H
+#  error "this header file must not be included directly"
+#endif
+
+// This is the implementation of Python atomic operations for MSVC if the
+// compiler does not support C11 or C++11 atomics.
+
+#include <intrin.h>
+
+
+static inline int
+_Py_atomic_add_int(int *address, int value)
+{
```

Comment:
Maybe add ``Py_BUILD_ASSERT(sizeof(int) == sizeof(long));``, since the cast is surprising.

Same remark for other casts below:

```
sizeof(uint8_t) == sizeof(char)
sizeof(uint16_t) == sizeof(short)
...
```